discovery-provider/src/challenges/create_new_challenges.py

```diff
@@ -2,18 +2,59 @@
 import logging
 import pathlib
 from os import path
+from typing import List, Optional, TypedDict
 
+from src.utils.config import shared_config
 from src.models import Challenge
 
 logger = logging.getLogger(__name__)
 
 
+class ChallengeJSON(TypedDict):
+    id: str
+    type: str
+    amount: str
+    active: bool
+    step_count: Optional[int]
+    starting_block: Optional[int]
+
+
+class OverrideChallengeJson(TypedDict):
+    id: str
+    amount: Optional[str]
+    active: Optional[bool]
+    step_count: Optional[int]
+    starting_block: Optional[int]
+
+
 def get_challenges_dicts():
     challenges_path = path.join(pathlib.Path(__file__).parent, "challenges.json")
+    challenges = []
     with open(challenges_path) as f:
         raw = f.read()
-        parsed = json.loads(raw)
-        return parsed
+        challenges: List[ChallengeJSON] = json.loads(raw)
+
+    # If we're in stage environment, set up overrides
+    if shared_config["discprov"]["env"] == "stage":
+        stage_challenges_path = path.join(
+            pathlib.Path(__file__).parent, "challenges.stage.json"
```

Comment:
not sure if you'd want to make this change, but you could do

f"challenges.{env}.json" instead of checking == "stage"

that would remove some headache some day if we add another env / dev / something

Reply:
nice